backend/sftp/sftp.go

```diff
@@ -44,6 +44,7 @@ const (
 
 var (
 	currentUser = env.CurrentUser()
+	savedpswd   = ""
```

Comment:
This variable needs to be stored inside `Fs` in case there is more than one sftp backend active

Reply:
OK, I'm not sure how to do that.  

Since `getPass()` is called via an SSH callback, I don't know how to get access to the Fs object.  All the other variables stored there appear to be called within the scope of a `(f *Fs)` function and so can be accessed by `f.variable`, but we don't seem to have that option, here!

I guess I could re-use `config.Pass` since we know the user hadn't specified a password ?  Or is that also a bad idea?

Reply:
I would make `getPass` into a method of `Fs`

```go
func (f *Fs) getPass() (string, error) {
```

Then you can use `f` within the function.

You can then pass the bound method `f.getPass` rather than `getPass` as the callback.

I hope that helps!

Reply:
Except `f` isn't defined in `NewFS` (where the `GetPass` method is referenced) because we don't _have_ an `Fs` object yet...

Reply:
I wonder if a simpler hack might be to define `f := &Fs{}` in `NewFS` so now I can refer to `f.getPass` in `NewFS`, and pass `f` as a parameter to a modified `func NewFsWithConnection(f *Fs,....)` and modify the initialiser to be `f.name =...` type stuff.  

+       return NewFsWithConnection(f, ctx, name, root, m, opt, sshConfig)

Let's test...

Hmm, that seems to work!

You're right it probably needs refactoring, but this'll may work around the problem.